### ISSUE TITLE
Typescript types no longer resolved from corresponding ".ts" file if import has ".js" extension from 0.21

### ISSUE THREAD

**Issue by jiby-gh** [tomblachut/svelte-intellij#240]
I have a typescript project setup that does not have a bundle step. Each typescript file is complied by web storm watches when it changes independently. And for the es6 imports to just work out of the box without needing transformations the code imports with ".js" extension instead of ".ts" extension.

Below is an example, where "webgpu.js" is imported instead of "webgpu.ts". 
![image](https://user-images.githubusercontent.com/52130524/127013844-28eb0e42-3b09-4b86-a4d2-b6b15750b175.png)

Typescript compiler seems to handle this properly as well as typescript files inside web storm. And it used to work in svelte files before I updated web storm 2021.2 and before version 0.21 of the plugin. It no longer works as can be seen in the screenshot. Not sure if it's an issue with changes in web storm 2021.2 or with version 0.21 of the plugin

**Comment by tomblachut** [comment]
Hello. Do I understand correctly, that you haven't used WebStrom 2021.2 with Svelte **0.20.0** (previous version)?

> Not sure if it's an issue with changes in web storm 2021.2 or with version 0.21 of the plugin

Please put everything from script tag into separate .ts file, uncheck TypeScript language service and see if something is different.
![image](https://user-images.githubusercontent.com/615013/127015364-e30a3bb4-1ed9-4ff0-be1e-60d49e0a8c74.png)

If there's still issue please provide small repro project.

**Comment by jiby-gh** [comment]
@tomblachut 

Unfortunately not, as I clicked upgrade plugins, as there was a bunch of plugins that I was that needed upgrade to work with 2021.2.

Well, putting everything in another ".ts" file also only works in svelte files if imported as ".ts" instead of ".js". 

I will setup a small repro project

**Comment by tomblachut** [comment]
> Well, putting everything in another ".ts" file also only works in svelte files if imported as ".ts" instead of ".js".

Not sure I understand, can you rephrase? This is a problem in Svelte plugin if the result is different in plain TS file, most problems are inherited by script tags inside Svelte files.

**Comment by jiby-gh** [comment]
@tomblachut find the repo with minimal repro here https://github.com/jiby-gh/svelte-intellij-issue-240.

If you just clone the file and open `src/NotOk.svelte`, it considers types properly, if you then do `npm install` which compiled `src/Logger.ts` to generate `src/Logger.js`, then `src/NotOk.svelte` will complain about `logger.log`.

Let me know if you are able to repro.

**Comment by tomblachut** [comment]
@jiby-gh Ok I see, log is unresolved. When I copy all contents of script tag to standalone TS, problem is exactly the same, therefore this is not a problem with Svelte. I can also confirm that is indeed worked in 2021.1.

![image](https://user-images.githubusercontent.com/615013/127027224-39aa1c3b-8b4e-4a5e-afe6-fd2012997260.png)

Please open separate issue in [YouTrack](https://youtrack.jetbrains.com/issues/WEB).

**Comment by jiby-gh** [comment]
interesting that it does not complain for you that type definitions are missing for `../node_modules/svelte/index.mjs` which i think it should have

**Comment by jiby-gh** [comment]
@tomblachut Okay, I just tried with the typescript language service disabled, and it complains in both. 

But is it unexpected behaviour, isn't the types in ts provided by the typescript language service, is it supposed to work without the typescript language service?

**Comment by jiby-gh** [comment]
@tomblachut unfortunately I can confirm that it does not happen with 0.20 of the plugin and web storm 2021.2. I just installed the old version from disk and tried. It works with that combo. So i think it has something to do with the changes in 0.21

![image](https://user-images.githubusercontent.com/52130524/127042417-a28359f5-60cd-4504-8ce7-5b12b20a7a76.png)

**Comment by tomblachut** [comment]
Oh, that's because 0.21 also fixes tsconfig resolution, so in 0.20 case bug was hidden by another one. Your problem still can't be fixed here, only upstream.

as for your question what does ts service do actually, it's really complicated and I can't answer this throughoutly. Types are evaluated in two places basically

**Comment by jiby-gh** [comment]
ah, okay, thanks for the clarification !
